File: mobile/app/src/main/java/com/data/remote/dto/UpdateTaskRequest.kt
package com.data.remote.dto


import com.google.gson.annotations.SerializedName

data class UpdateTaskRequest(
    @SerializedName("taskId")
    val taskId: String,
    @SerializedName("title")
    val title: String?,
    @SerializedName("description")
    val description: String?,
    @SerializedName("dueDate")
    val dueDate: String?,
    @SerializedName("completed")
    val isCompleted: Boolean?
)

File: api/Taskolith.API/Tasks/Requests/UpdateTaskRequest.cs
using Taskolith.API.Data.Types;

namespace Taskolith.API.Tasks.Requests;

public class UpdateTaskRequest {
    public string? Title { get; init; }
    public string? Description { get; init; }
    public DateTime? DueDate { get; init; }
    public int? Order { get; init; }
    public Priority? Priority { get; init; }
    public bool? IsCompleted { get; init; }
    public Guid? KanbanColumnId { get; init; }
};